includes/admin.inc

```diff
@@ -704,6 +704,60 @@ function islandora_solr_admin_settings($form, &$form_state) {
 }
 
 /**
+ * Confirmation for reset solr settings.
+ *
+ * @param array $form
+ *   Array containing all form elements.
+ * @param array $form_state
+ *   The form_state array of the form. Includes the values.
+ *
+ * @return array
+ *   Rendered confirmation form.
+ */
+function islandora_solr_admin_settings_default_confirm_form(array $form, array &$form_state) {
+  $form = confirm_form($form,
+    t('Confirm settings reset to default'),
+        'admin/islandora/search/islandora_solr/settings',
+    t('Confirm reset settings to default, this cannot be undone.'),
+    t('Continue'),
+    t('Cancel')
+  );
+  return $form;
+}
+
+/**
+ * If user confirms reset solr settings delete settings.
+ *
+ * Iterate through an array of Solr settings and remove the present values,
+ * redirect user back to settings page.
+ *
+ * @param array $form
+ *   Array containing all form elements.
+ * @param array $form_state
+ *   The form_state array of the form. Includes the values.
+ */
+function islandora_solr_admin_settings_default_confirm_form_submit(array $form, array &$form_state) {
+
+  // Solr Settings to be reset.
+  $values = array('islandora_solr_result_fields', 'islandora_solr_limit_result_fields', 'islandora_solr_num_of_results',
+    'islandora_solr_search_navigation', 'islandora_solr_sort_fields', 'islandora_solr_facet_fields', 'islandora_solr_facet_min_limit',
+    'islandora_solr_facet_soft_limit', 'islandora_solr_facet_max_limit', 'islandora_solr_search_fields', 'islandora_solr_search_boolean',
+    'islandora_solr_allow_preserve_filters', 'islandora_solr_human_friendly_query_block', 'islandora_solr_namespace_restriction',
+    'islandora_solr_base_query', 'islandora_solr_base_advanced', 'islandora_solr_base_sort', 'islandora_solr_base_filter',
+    'islandora_solr_query_fields', 'islandora_solr_use_ui_qf', 'islandora_solr_content_model_field', 'islandora_solr_datastream_id_field',
+    'islandora_solr_object_label_field', 'islandora_solr_member_of_field', 'islandora_solr_member_of_collection_field', 'islandora_solr_debug_mode',
+    'islandora_solr_tabs__active_tab', 'islandora_solr_primary_display_table', 'islandora_solr_secondary_display');
+  foreach ($values as $value) {
+    variable_del($value);
+  }
+  variable_del('islandora_solr_primary_display');
+  variable_del('islandora_solr_request_handler');
```

Comment:
I'm a bit concerned about keeping all these variable names in two places now that they're no longer in the `$form_state`; this would require us to maintain the list of variables both here _and_ in the [hook_install() implementation](https://github.com/Islandora/islandora_solr_search/blob/22351cbbf9c0adba0eaa7d3a62d85d8565a231ca/islandora_solr.install#L45-L85) if we ever make any changes to it. Since it's already our protocol to add/remove variables to/from `hook_install()`, could we instead break out the variable deletion to a separate function in `islandora_solr.install` and call it from both `hook_install()` and `islandora_solr_admin_settings_default_confirm_form_submit()`?